### PR TITLE
Allow variadic params for the store

### LINKED ISSUE #19: How to Use Etcd for Distributed KV Storage And Let the Dev choose Consistency/Speed
Problem: We don't understand how to utilize etcd for our distributed key-value storage yet. 

Research and design how a system would work for this. Take into account our design for service discovery and find a way that compliments both. 

Examples of things we are unsure about:
- If we update a value on our local etcd, how does it get propagated to the rest of the nodes? Or do we always have to send writes to the leader?
- If we attempt to read a key on our local etcd and the value isn't there will it redirect it? Or do we have to request the leader for reads?

### PR DESCRIPTION
**Pull request by dipshit**
closes #19 

This PR copies the OpOptions functions from [here](https://github.com/etcd-io/etcd/blob/v3.3.18/clientv3/op.go) and brings them in to our own package to wrap. 
Why?
* This way the developer won't need to import `clientv3` to configure our package.
* The work was already done for us by etcd.
* It makes the API the same in case our docs are bad

I didn't add many tests for these wrappers, as they are just calling the `clientv3` code.

### CODE REVIEW

**Comment by JackyChiu** [comment]
Some questions and thoughts:

1. Do we need to wrap all the options? I think we're in a position with the library where we can limit the scope of the possible options.
2. Should we be creating our own option type? This would provide the flexibility of added a `WithContext` and a `WithTimeout` option for example. 
3. It is now up to the user to append a "/" to the key and add a `WithPrefix` option if they want to treat the key as a directory? Maybe we could provide another option that does both?

**Comment by dipshit** [comment]
1. Ya I will delete unnecessary options. I just copied them all in to make a PR to see people's opinions on this solution.
2. There is no reason to because I want the ctx to be passed in the same way that the etcd API has it. Is a `WithTimeout` useful at all if they can pass the ctx in? If there are other configuration options wanted, then making our own option type is useful, but for now I see no reason.
3. The key can be treated as a directory without `/` being appended, as long as `WithPrefix()` is used. I removed the appending because of the case I encountered in the tests:
 * Put `raccoon`
 * Put `raccoon2`
 * Get `racoon` `WithPrefix()`
 * Only the value of `raccoon` is returned, because the keys are stored as `raccoon/` and `raccoon2/`
Appending the `/` without telling the developer is unfriendly, I would expect the above example to give me the values for `raccoon` and `raccoon2`, but with the `/` it does not.
I don't see a reason to add something like a `WithPrefixAndSlash()` when the developer can avoid using `WithPrefix()` in every call, and end keys without `/`.

**Comment by JackyChiu** [comment]
1. 👍 
2. 👍 
3. 
> The key can be treated as a directory without / being appended

If this is true, then it looks good to me. Otherwise, I am concerned about this case, where all the keys would be returned when only the ones under "jobs/" should be. 

- `put "jobs/raccoon1" val1`
- `put "jobs/raccoon2" val2`
- `put "jobs_old/raccoon99" val99`
- `get "jobs" with prefix`

I ran into this issue with the calculator example where we had a service `calculator` and `calculator_client`.

**Comment by dipshit** [comment]
3. I did not consider that. That is a good reason to add an option for a / suffix and Prefixed search. I can do this without making our own option type too 😎

File: cluster/store_config.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+// WithSerializable makes 'Get' request serializable. By default,
+// it's linearizable. Serializable requests are better for lower latency
+// requirement.
+func WithSerializable() clientv3.OpOption {
+	return clientv3.WithSerializable()
+}
+
+// WithKeysOnly makes the 'Get' request return only the keys and the corresponding
+// values will be omitted.
+func WithKeysOnly() clientv3.OpOption {
```
Comment (dipshit) [inline cluster/store_config.go]: This reminded me that we can't serve just keys. #59

**Comment by dipshit** [comment]
I did not add an option to append a '/' and search with prefix, because the developer can append a '/' and call with WithPrefix() themselves. This solution covers all issues we talked about in the pr comments

**Pull request merged**